### PR TITLE
Improved ProgramYear Stewards

### LINKED ISSUE #2528: Refactor program year stewards
This area hasn't been touched for a while and has a few issues:
1) No spinner when saving
2) A LOT of network requests get fired even if nothing was changed
3) Should be collapsed
4) Is using DS.PromiseArray instead of promises

probably more things, but those are the immediate concerns.

### PR DESCRIPTION
**Pull request by jrjohnson**
This set of interactions was full of old patterns and old code and is now ultra modern and shiny!  Hopefully this cleanup allows us to move to ember-data 2.11 and improves the user experience slightly.

Fixes #2528

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview ready!

Built with commit 1cc3cc82156b7e6be3dd44710df5a8388200d3b0

https://deploy-preview-2534--ilios-frontend.netlify.com

**Comment by saschaben** [comment]
Clickety click looks good. Like the spinners in the list boxes!

**Comment by jrjohnson** [comment]
@stopfstedt this is ready for review, the failing test is that I forgot to update translations and will have those soon.  I'm guessing there will probably be some questions about some of the new stuff I experimented here as well.  And maybe the weird hack I put in and labeled as such.

File: app/components/collapsed-stewards.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      const departmentId = isPresent(department)?department.get('id'):0;
+      return {
+        schoolId: school.get('id'),
+        schoolTitle: school.get('title'),
+        departmentId
+      };
+    });
+    const schools = stewardObjects.uniqBy('schoolId');
+    const schoolData = schools.map(obj => {
+      const deparrments = stewardObjects.filterBy('schoolId', obj.schoolId);
```
Comment (stopfstedt) [inline app/components/collapsed-stewards.js]: typo in variable name, should be `departments`.

File: app/components/collapsed-stewards.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      delete obj.departmentId;
+      obj.departmentCount = deparrments.length;
```
Comment (stopfstedt) [inline app/components/collapsed-stewards.js]: likewise, should be `departments`.

File: app/components/detail-steward-manager.js [hunk 3]
```diff
@@ -1,152 +1,163 @@
...
+  availableSchools: computed('selectedDepartments.[]', 'selectedSchools.[]', 'allSchools.[]', async function(){
+    const allSchools = await this.get('allSchools');
+    const selectedDepartments = await this.get('selectedDepartments');
+    const selectedSchools = await this.get('selectedSchools');
+
+    const selectedDepartmentIds = selectedDepartments.mapBy('id');
+    const selectedSchoolIds = selectedSchools.mapBy('id');
+    const unselectedSchoolsAndSchoolsWithUnselectedDepartments = await filter(allSchools.toArray(), async school => {
+      const departments = await school.get('departments');
+      const unlselectedDepartments = departments.filter(department => {
```
Comment (stopfstedt) [inline app/components/detail-steward-manager.js]: typo in variable name, should be `unselectedDepartments`.

File: app/components/detail-steward-manager.js [hunk 4]
```diff
@@ -1,152 +1,163 @@
...
+        return !selectedDepartmentIds.includes(department.get('id'));
       });
-      this.sendAction('add', steward);
-    },
-    removeSchool: function(school){
-      let steward = this.get('stewards').find(steward => {
-        return steward.get('school.id') === school.get('id') && !steward.get('department.content');
+      return !selectedSchoolIds.includes(school.get('id')) || isPresent(unlselectedDepartments);
```
Comment (stopfstedt) [inline app/components/detail-steward-manager.js]: same here, should be `unselectedDepartments`.

File: app/components/collapsed-stewards.js (hunk 1 above)
Comment (jrjohnson) [inline app/components/collapsed-stewards.js]: fixed.  Thanks.

**Pull request merged**